src/main/kotlin/app/hashers/CodeLongevity.kt

```diff
@@ -83,30 +92,31 @@ class CodeLongevity(repoPath: String, tailRev: String) {
      */
     var codeLines: MutableList<CodeLine> = mutableListOf()
 
-    init {
+    fun update() {
         compute()
-    }
 
-    // TODO(alex) debugging, remove it
-    fun ohNoDoesItReallyWork(email: String) {
-        var sum: Long = 0
-        var total: Long = 0
+        // TODO(anatoly): Add emails from server or hashAll.
+        val emails = hashSetOf(localRepo.author.email)
+
+        val sum: MutableMap<String, Long> = emails.associate { Pair(it, 0L) }
+                                                  .toMutableMap()
+        val total: Int = codeLines.size
         for (line in codeLines) {
-            val author = line.from.commit.getAuthorIdent()
-            if (author.getEmailAddress() != email) {
+            val email = line.from.commit.authorIdent.emailAddress
+            if (!emails.contains(email)) {
```

Comment:
We also need to compute average for entire repo. Remember the chart?

Reply:
Will be in next pr very-very soon.